src/qibo/backends/numpy.py

```diff
@@ -638,7 +638,7 @@ def samples_to_binary(self, samples, nqubits):
     def samples_to_decimal(self, samples, nqubits):
         qrange = self.np.arange(nqubits - 1, -1, -1, dtype="int32")
         qrange = (2**qrange)[:, self.np.newaxis]
-        return self.np.matmul(self.to_numpy(samples), qrange)[:, 0]
+        return self.np.matmul(self.np.vstack(self.to_numpy(samples)), qrange)[:, 0]
```

Comment:
Just to understand, is there a reason for this further `vstack`?

Reply:
I remember this was failing otherwise, probably with `cupy`.

Reply:
Ok I double checked this and, indeed, it's not necessary, I honestly don't remember when I was getting the failure...